maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -406,6 +443,75 @@ public SessionBuilder newRepositorySessionBuilder(MavenExecutionRequest request)
         return session;
     }
 
+    private VersionFilter buildVersionFilter(Map<Object, Object> configProps) {
+        ArrayList<VersionFilter> filters = new ArrayList<>();
+        String filterExpression = (String) configProps.get(MAVEN_VERSION_FILTERS);
+        if (filterExpression != null) {
+            List<String> expressions = Arrays.stream(filterExpression.split(","))
+                    .filter(s -> s != null && !s.trim().isEmpty())
+                    .collect(Collectors.toList());
+            for (String expression : expressions) {
+                if ("h".equals(expression)) {
+                    filters.add(new HighestVersionFilter());
+                } else if (expression.startsWith("h(") && expression.endsWith(")")) {
+                    int num = Integer.parseInt(expression.substring(2, expression.length() - 1));
+                    // MRESOLVER-450
+                    // filters.add(new HighestVersionFilter(num));
```

Comment:
I think it's https://github.com/apache/maven-resolver/pull/401 / MRESOLVER-460.
This PR should be moved to draft to avoid anyone forgetting it's not finished and we need next resolver alpha.